packages/loader/container-loader/src/protocolTreeDocumentStorageService.ts

```diff
@@ -38,9 +48,14 @@ export class ProtocolTreeStorageService implements IDocumentStorageService, IDis
 		summary: ISummaryTree,
 		context: ISummaryContext,
 	): Promise<string> {
-		return this.internalStorageService.uploadSummaryWithContext(
-			this.addProtocolSummaryIfMissing(summary),
-			context,
-		);
+		if (this.shouldSummarizeProtocolTree()) {
+			this.logger.sendTelemetryEvent({ eventName: "summarizeProtocolTreeEnabled" });
+			return this.internalStorageService.uploadSummaryWithContext(
```

Comment:
if you move logging call into addProtocolSummaryIfMissing(), then this code can be much simpler:

```ts
return this.internalStorageService.uploadSummaryWithContext(
   this.shouldSummarizeProtocolTree() ? this.addProtocolSummaryIfMissing(summary) : summary,
   context);
```

Reply:
Good point. FIxed.